dali/base/dacoven.cpp

```diff
@@ -581,23 +586,30 @@ class CCovenServer: public CCovenBase
     {
         if (num==0)
             return 0;
+
+        CDaliUidAllocator *uidAllocator = nullptr;
+
         SocketEndpoint foreignnode;
         if (_foreignnode&&!_foreignnode->isNull()) {
             foreignnode.set(*_foreignnode);
             if (foreignnode.port==0)
                 foreignnode.port = DALI_SERVER_PORT;
+
+            uidcrit.enter();
```

Comment:
Why not use a CriticalBlock?

Reply:
no good reason, just that this was pre-existing code (just moved a bit).
Could change?

Reply:
If you are going to refactor may as well do it properly